Assets/Scripts/RoadControl.cs

```diff
@@ -0,0 +1,616 @@
+            seg.Curve = rdCurve.curveIntensity - rdCurve.curveIntensity * (-Mathf.Cos((i / rdCurve.numEnterSegments) * Mathf.PI) / 2 + 0.5f);
+            seg.Y = startY + (endY - startY) * ((-Mathf.Cos(((i + rdCurve.numEnterSegments + rdCurve.numHoldSegments) / total) * Mathf.PI) / 2) + 0.5f);
+            seg.SpriteVariation = curRoadStripSprite;
+            roadSegments.Add(seg);
+        }
+    }
+
+    private void CheckRumbleStripCounter() {
+        if (rumbleStripCounter > numSegsPerRumble) {
+            curRoadStripSpriteIndex = (curRoadStripSpriteIndex + 1) % roadStripSprites.Count;
+            curRoadStripSprite = roadStripSprites[curRoadStripSpriteIndex];
+            rumbleStripCounter = 1;
+        }
+    }
+
+    private void IncrementSegmentsAndRumbleStrips() {
+        curAddedSegmentZ += roadSegmentLength;
+        rumbleStripCounter++;
+    }
+
+    private void ManagePlayerPosition() {
+
+        curPlayerTurning = maxTurning;
+
+        speedometerNeedle.rectTransform.rotation = Quaternion.Euler(0, 0, curPlayerSpeed * speedometerRotationFactor);
+        speedometerReading.text = ((int)curPlayerSpeed / 12).ToString();
+
+        if (canDrive) {
+            if (InputMapper.inputMapper[(int)InputMapper.CONTROLS.up] != null && InputMapper.inputMapper[(int)InputMapper.CONTROLS.up].IsPressed()) {
+                if (curPlayerSpeed < topSpeed) { curPlayerSpeed += acceleration * Time.deltaTime; }
+                else { curPlayerSpeed = topSpeed; }
+            }
+            else {
+                if (curPlayerSpeed > 0) { curPlayerSpeed -= deceleration * Time.deltaTime; }
+                else { curPlayerSpeed = 0.0f; }
+            }
+
+            if (InputMapper.inputMapper[(int)InputMapper.CONTROLS.left] != null && InputMapper.inputMapper[(int)InputMapper.CONTROLS.left].IsPressed()) {
+
+                if (InputMapper.inputMapper[(int)InputMapper.CONTROLS.action] != null && InputMapper.inputMapper[(int)InputMapper.CONTROLS.action].IsPressed()) {
+                    playerCarSprite.sprite = playerCarHardLeft;
+                    if (Mathf.Abs(playerCar.position.x) < screenHalfWidth) {
+                        playerCar.Translate(-maxTurning * Time.deltaTime, 0, 0);
+                    }
+                }
+                else {
+                    playerCarSprite.sprite = playerCarLeft;
+                    if (Mathf.Abs(playerCar.position.x) < screenHalfWidth) {
+                        playerCar.Translate(-maxTurning / 2 * Time.deltaTime, 0, 0);
+                    }
+                }
+            }
+            else if (InputMapper.inputMapper[(int)InputMapper.CONTROLS.right] != null && InputMapper.inputMapper[(int)InputMapper.CONTROLS.right].IsPressed()) {
+
+                if (InputMapper.inputMapper[(int)InputMapper.CONTROLS.action] != null && InputMapper.inputMapper[(int)InputMapper.CONTROLS.action].IsPressed()) {
+                    playerCarSprite.sprite = playerCarHardRight;
```

Comment:
I'm sort of embarrassed by my old code's lack of robustness.
My new code addresses this much better so as a quick fix:

`if (InputMapper.inputMapper[(int)InputMapper.CONTROLS.up] != null && InputMapper.inputMapper[(int)InputMapper.CONTROLS.up].IsPressed()) {`

should become:
`if (InputMapper.inputMapper[(int)InputMapper.CONTROLS.up]) {`

after you get my next update, that is.